app/src/main/java/com/example/internetshop/presentation/adapters/ProductsCategoryAdapter.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.internetshop.presentation.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.internetshop.databinding.ItemProductCategoryBinding
+import com.example.internetshop.domain.data.model.product.SimpleProduct
+import com.example.internetshop.presentation.adapters.holder.ProductCategoryViewHolder
+
+class ProductsCategoryAdapter(val clickListener: (SimpleProduct) -> Unit) :
+    RecyclerView.Adapter<ProductCategoryViewHolder>() {
+    var productList: MutableList<SimpleProduct> = mutableListOf()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProductCategoryViewHolder {
+        return ProductCategoryViewHolder(
+            ItemProductCategoryBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: ProductCategoryViewHolder, position: Int) {
+        holder.bind(productList[position])
+        holder.itemView.setOnClickListener {
+            clickListener(productList[position])
+        }
+    }
+
+    override fun getItemCount(): Int {
```

Comment:
In this method you only return value, so would be better to use body style here

Reply:
fixed